commands/owner/reload.js

```diff
@@ -0,0 +1,172 @@
+const { CommandInteraction, EmbedBuilder } = require('discord.js');
+const fs = require('fs').promises;
+const path = require('path');
+const config = require('../../config');
+const { REST, Routes } = require('discord.js');
+
+module.exports.data = {
+  name: 'reload',
+  description: 'Reload all functions, commands, events, and discord-player',
+  type: 1,
+  options: [
+    {
+      name: 'deploy',
+      description: 'Deploy new commands after reloading',
+      type: 5, // BOOLEAN
+      required: false,
+    },
+  ],
+  integration_types: [0],
+  contexts: [0],
+  owner: true,
+};
+
+/**
+ * @param {CommandInteraction} interaction
+ */
+module.exports.execute = async interaction => {
+  if (!config.OwnerID.includes(interaction.user.id)) return;
+
+  await interaction.deferReply();
+
+  const client = interaction.client;
+  const shouldDeploy = interaction.options.getBoolean('deploy') ?? false;
+  const reloadResults = await reloadAll(client);
+
+  if (shouldDeploy) {
+    const deployResults = await deployCommands(client);
+    reloadResults.deployed = deployResults;
+  }
+
+  const embed = new EmbedBuilder()
+    .setTitle('Reload Results')
+    .setColor('Green')
+    .addFields(
+      Object.entries(reloadResults).map(([key, value]) => ({
+        name: key.charAt(0).toUpperCase() + key.slice(1),
+        value: formatReloadResult(value),
+        inline: true,
+      }))
+    )
+    .setTimestamp();
+
+  await interaction.editReply({ embeds: [embed] });
+};
+
+async function reloadAll(client) {
+  const reloadFunctions = {
+    functions: ['../../functions', client.functions],
+    commands: ['../../commands', client.commands],
+    events: ['../../events', null, reloadEvent],
+    discordPlayer: ['../../discord-player', null, reloadDiscordPlayerEvent],
+  };
+
+  const results = {};
+  for (const [key, [dirPath, collection, additionalAction]] of Object.entries(reloadFunctions)) {
+    if (key === 'events') client.removeAllListeners();
+    if (key === 'discordPlayer' && client.player) {
```

Comment:
client.player không có thực vì thực tế player không được gắn vào client mà nó sử dụng https://discord-player.js.org/guide/faq/how-to-access-player 
Lệnh này để sau đi nào tạo shard thì cứ shutdown nó tự khởi động lại thoi